src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -159,19 +154,11 @@ public boolean makeOTPRequestAndUpdateMatchingItinerary() {
             // Generate the appropriate OTP query params for the trip for the current check by replacing the date query
             // parameter with the appropriate date.
             Map<String, String> params = trip.parseQueryParams();
-
-            // building a new list by copying all values, except for the date which is set to the target date
-            List<NameValuePair> newParams = new ArrayList<>();
-            newParams.add(new BasicNameValuePair("date", targetDate));
-            for (Map.Entry<String, String> param : params.entrySet()) {
-                if (!param.getKey().equals("date")) {
-                    newParams.add(new BasicNameValuePair(param.getKey(), param.getValue()));
-                }
-            }
-            otpDispatcherResponse = OtpDispatcher.sendOtpPlanRequest(URLEncodedUtils.format(newParams, UTF_8));
+            params.put("date", targetDate);
```

Comment:
We should probably use `ItineraryUtils#DATE_PARAM` here.

Reply:
Fixed in 1987213b6754ffa98f4748820b628be64f9443f4